syntax = "proto3";

package stock;

service StockPredictor {
  rpc Predict(PredictRequest) returns (PredictResponse);
}

message PredictRequest {
  repeated string tickers = 1;
}

message StockPrediction {
  string ticker = 1;
  float predicted_price = 2;
}

message PredictResponse {
  repeated StockPrediction predictions = 1;
  bytes plot_image = 2; // Matplotlib PNG image as bytes
}
